operator/controllers/manifest_controller_test.go

```diff
@@ -42,60 +36,165 @@ func createManifest() *v1alpha1.Manifest {
 					Name: "nginx-stable",
 				},
 			},
-		},
+		})
+		Expect(k8sClient.Create(ctx, manifestObj)).Should(Succeed())
+		Eventually(getManifestState(client.ObjectKeyFromObject(manifestObj)), 5*time.Minute, 250*time.Millisecond).
+			Should(BeEquivalentTo(v1alpha1.ManifestStateReady))
+		Expect(k8sClient.Delete(ctx, manifestObj)).Should(Succeed())
+		Eventually(getManifest(client.ObjectKeyFromObject(manifestObj)), 5*time.Minute, 250*time.Millisecond).
+			Should(BeTrue())
+		return true
 	}
 }
 
-func getManifestState(key client.ObjectKey) func() string {
-	return func() string {
+func createManifestWithOCI() func() bool {
```

Comment:
these tests are pretty big to read and they return a bool func. Could we somehow make the methods a bit more concise/smaller so its easier to follow through? I cannot put the entire method on my screen at once

Reply:
I agree, I refactored tests to make them more readable, but they're still not the appropriate size IMO.